### PR TITLE
filter_kubernetes: extract docker_id and container_hash from Kubernetes API

### PR DESCRIPTION
**Pull request by donbowman**
Extract docker_id and container_hash from Kubernetes API

In the systemd journal input case, the 'docker_id' and 'container_hash'
fields were extracted from the journald. In the 'tail' input case these
were missing. Since they are both available from the Kubernetes api,
and since we have already fetched the data from this api, use it when
present from the `status.containerStatuses` section.

Signed-off-by: Don Bowman <don@agilicus.com>

### CODE REVIEW

File: plugins/filter_kubernetes/kube_meta.c [hunk 1]
```diff
@@ -290,6 +290,84 @@ static void cb_results(unsigned char *name, unsigned char *value,
...
+                                   msgpack_object status)
+{
+    int i;
+    msgpack_object k, v;
+    int docker_id_len = 0;
+    int container_hash_len = 0;
+    const char *container_hash;
+    const char *docker_id;
+    int name_found = FLB_FALSE;
+    /* Process status/containerStatus map for docker_id, container_hash */
```
Comment (edsiper) [inline plugins/filter_kubernetes/kube_meta.c]: please document here how the payload looks like to have an easy understanding of the lookup code.

Comment (donbowman) [reply plugins/filter_kubernetes/kube_meta.c]: done

**Comment by edsiper** [comment]
thanks for the changes, in order to merge the commits please rename the commit subjects (prefix separated by a colon):

https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#commit-changes

**Comment by donbowman** [comment]
> thanks for the changes, in order to merge the commits please rename the commit subjects (prefix separated by a colon):
> 
> https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#commit-changes

you should do a squash when you accept a PR. This will squash to the original commit message which matches your format I think.

**Pull request merged**